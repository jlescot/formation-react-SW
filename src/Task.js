import React from 'react';

function Task(props) {
  return (
    <li onClick={props.handleToggleStatus} style={{ cursor: 'pointer', lineHeight: '30px' }}>
      {props.done ? 'DONE' : 'TODO'} - {props.name}
    </li>
  );
}

export default Task;
