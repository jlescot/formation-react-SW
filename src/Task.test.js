import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Task from './Task';

describe('<Task />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Task />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('can display a DONE Task', () => {
    const wrapper = shallow(<Task done name="Task1" />);
    expect(wrapper.text()).toEqual('DONE - Task1');
  });

  it('can display a TODO Task', () => {
    const wrapper = shallow(<Task name="Task2" />);
    expect(wrapper.text()).toEqual('TODO - Task2');
  });

  it('calls handleToggleStatus on click', () => {
    const handleToggleStatus = jest.fn();
    const wrapper = shallow(<Task handleToggleStatus={handleToggleStatus} name="Task3" />);
    const task = wrapper.find('li');

    task.simulate('click');
    expect(handleToggleStatus).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const handleToggleStatus = jest.fn();
    const wrapper = renderer.create(<Task handleToggleStatus={handleToggleStatus} name="Task4" />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
