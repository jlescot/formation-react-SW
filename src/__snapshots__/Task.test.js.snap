// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Task /> matches snapshot 1`] = `
<li
  onClick={[MockFunction]}
  style={
    Object {
      "cursor": "pointer",
      "lineHeight": "30px",
    }
  }
>
  TODO
   - 
  Task4
</li>
`;
